adagucserverEC/CDBAdapterPostgreSQL.cpp

```diff
@@ -356,181 +374,104 @@ CDBStore::Store *CDBAdapterPostgreSQL::getFilesAndIndicesForDimensions(CDataSour
+                                   dimName, tableName, dimName, dimVals[1].c_str());
       }
-      subQuery.printconcat("ORDER BY %s DESC limit %d)a%d ", netCDFDimName.c_str(), limit, i);
-      // subQuery.printconcat("ORDER BY %s DESC )a%d ",netCDFDimName.c_str(),i);
-    } else {
-      subQuery.printconcat("ORDER BY %s DESC)a%d ", netCDFDimName.c_str(), i);
+      delete[] dimVals;
     }
-    // subQuery.printconcat("ORDER BY %s DESC)a%d ",netCDFDimName.c_str(),i);
-    if (i < dataSource->requiredDims.size() - 1) subQuery.concat(",");
-    queryOrderedDESC.concat(&subQuery);
-  }
-  //  CDBDebug("%s",queryOrderedDESC.c_str());
-#ifdef CDBAdapterPostgreSQL_DEBUG
-  CDBDebug("%s", queryOrderedDESC.c_str());
-#endif
-  // Join by path
-  if (dataSource->requiredDims.size() > 1) {
-    queryOrderedDESC.concat(" where a0.path=a1.path");
-    for (size_t i = 2; i < dataSource->requiredDims.size(); i++) {
-      queryOrderedDESC.printconcat(" and a0.path=a%d.path", i);
+    if (!whereStatement.empty()) {
+      query.printconcat("AND (%s) ", whereStatement.c_str());
     }
-  }
-#ifdef CDBAdapterPostgreSQL_DEBUG
-  CDBDebug("%s", queryOrderedDESC.c_str());
-#endif
 
-  if (timeValidationError == true) {
-    if ((CServerParams::checkDataRestriction() & SHOW_QUERYINFO) == false) queryOrderedDESC.copy("hidden");
-    CDBError("queryOrderedDESC fails regular expression: '%s'", queryOrderedDESC.c_str());
-    return NULL;
+    delete[] requestedDimVals;
   }
 
-  query.print("select distinct * from (%s)T order by ", queryOrderedDESC.c_str());
-  query.concat(&dataSource->requiredDims[0]->netCDFDimName);
+  // Order and limit query
+  query.printconcat("ORDER BY %s", dataSource->requiredDims[0]->netCDFDimName.c_str());
   for (size_t i = 1; i < dataSource->requiredDims.size(); i++) {
-    query.printconcat(",%s", dataSource->requiredDims[i]->netCDFDimName.c_str());
+    query.printconcat(", %s", dataSource->requiredDims[i]->netCDFDimName.c_str());
   }
+  query.printconcat(" LIMIT %d", limit);
 
   // Execute the query
-
-  // writeLogFile3(query.c_str());
-  // writeLogFile3("\n");
-// values_path = DB.query_select(query.c_str(),0);
-#ifdef CDBAdapterPostgreSQL_DEBUG
-  CDBDebug("%s", query.c_str());
-#endif
-
   CDBStore::Store *store = NULL;
   try {
     store = DB->queryToStore(query.c_str(), true);
   } catch (int e) {
     if ((CServerParams::checkDataRestriction() & SHOW_QUERYINFO) == false) query.copy("hidden");
-    setExceptionType(InvalidDimensionValue);
+    // The method "setExceptionType" always sets the HTTP statuscode to 404. This breaks the test:
+    //    test_WMSGetCapabilities_no_error_on_existing_dataset_misconfigured_layer
+    // SQLite does not call setExceptionType when the query fails.
+    // setExceptionType(InvalidDimensionValue);
```

Comment:
A test was added in https://github.com/KNMI/adaguc-server/pull/387 which breaks when ran against postgresql. The test asserts that Adaguc should return 200 even if one of the layers is broken.

During the test, a non-existing table gets queried. In SQLite this exception is caught without calling `setExceptionType`. In PSQL we do set `setExceptionType`, which in turn always sets the response HTTP status code to 404.

I think it is better if the database class does not bother with HTTP status codes, maybe this behavior is better handled somewhere else. If we want Adaguc to always respond with a 200, even if a table is missing, we should remove this call.

Reply:
Agree.

Reply:
I'll remove `    // setExceptionType(InvalidDimensionValue);`